encoder/src/main/java/com/pedro/encoder/input/gl/render/ScreenRender.java

```diff
@@ -78,11 +78,12 @@ public void initGl(Context context) {
   }
 
   public void draw(int width, int height, boolean keepAspectRatio, int mode, int rotation,
-      boolean isPreview) {
+      boolean isPreview, boolean isCameraFront, boolean flipStreamHorizontal, boolean flipStreamVertical) {
     GlUtil.checkGlError("drawScreen start");
 
-    if (mode == 2 || mode == 3) { //stream rotation is enabled
-      SizeCalculator.updateMatrix(rotation, width, height, isPreview, isPortrait, MVPMatrix);
+    if (mode == 2 || mode == 3 || flipStreamHorizontal || flipStreamVertical) { //stream rotation is enabled
```

Comment:
Do we need those flips in the if check?

Reply:
I guess we need, because the whole flipping happens in `updateMatrix` method, which originally executes only when `aspectRatioMode` is 2 or 3. We're not changing those modes here, so additional condition is needed to execute this method.

Reply:
The other option would be to remove this if completely and move it to the `updateMatrix` method itself.